docs/building-on-lisk/deploying-smart-contract/with-Hardhat.md

```diff
@@ -216,6 +216,70 @@ npx hardhat compile
 
 After successful compilation, you should see a new folder `artifacts/`, which contains the [compilation artifacts](https://hardhat.org/hardhat-runner/docs/advanced/artifacts).
 
+## Testing the contract
+To test our NFT contract, we’ll use:
+
+- **Hardhat Network** – A local Ethereum network for development.
+It comes built-in with Hardhat, and it's used as the default network.
+- **Ethers.js** – To interact with the contract.
+- **Mocha** – For test runner and assertions.
+
+###  Setting up the test file
+Create a new directory called `test` inside our project root directory and create a new file in there called `NFT.js` and add the following code:
+
+```js title="test/NFT.js"
+const { expect } = require("chai");
+
+describe("NFT", function () {
+  let nftToken;
+
+  beforeEach(async () => {
+    // Deploy contract
+    const NFT = await ethers.getContractFactory("NFT");
+    nftToken = await NFT.deploy();
+  });
+
+  it("Should allow to mint a new NFT", async function () {
+    const [owner] = await ethers.getSigners();
+    // Before minting of the NFT, the account balance for this NFT should be zero.
+    expect(await nftToken.balanceOf(owner.address)).to.equal(0);
+    // Mint the NFT
+    await nftToken.mint(owner.address);
+    // After minting of the NFT, the account balance for this NFT should be 1 for the account that minted the token.
+    expect(await nftToken.balanceOf(owner.address)).to.equal(1);
+  });
+});
+```
+
+First, we import the `expect` function from the [Chai](https://www.chaijs.com/) library, to be able to use it in our unit test for the contract.
+
+If you set up your project to use the hardhat toolbox like explained in step [Creating a project](#creating-a-project), you already have ethers available in your project out of the box.
+Otherwise you can install the ethers plugin for Hardhat as described [here](https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-ethers).
```

Comment:
```suggestion
Otherwise, you can install the ethers plugin for Hardhat as described in the [hardhat-ethers](https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-ethers)' reference page.
```